server/mediaserver.go

```diff
@@ -187,7 +194,23 @@ func createRTMPStreamIDHandler(s *LivepeerServer) func(url *url.URL) (strmID str
 			mid, key = parseManifestID(resp.ManifestID), resp.StreamKey
 			// Process transcoding options presets
 			if len(resp.Presets) > 0 {
-				presets = parsePresets(resp.Presets)
+				profiles = parsePresets(resp.Presets)
+			}
+
+			if len(resp.Profiles) > 0 {
+				for _, profile := range resp.Profiles {
+					prof := ffmpeg.VideoProfile{
```

Comment:
Maybe if `profile.Name` is empty we can generate a name, eg `fmt.Sprintf(%dx%d_%d", width, height, bitrate)` . Had this issue when testing and forgot to add a name. Would be good to that behavior to other helpers as well, including `makeFFmpegVideoProfiles` and `FFmpegProfiletoNetProfile` . (And tests for those as well.)

Reply:
good catch, I'll add that + tests!